source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -75,22 +81,42 @@ public synchronized static TelemetryClient getTelemetryClient() {
     }
 
     public static void sendMetric(final String name, final double value) {
+        if (sendingTelemetryDisabled) {
```

Comment:
Another random thought...  If TelemetryClient is an interface instead of an implementation, it would be so much easier to do this - we could just return a no-op client from the getTelemetryClient() function and put all this logic in a single place.  However since TelemetryClient is an implementation, we are forced to short circuit the logic in many places.  I think this example really show case the "refactorablity" of programming against interface vs programming against implementation...  Oh well...